### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by chshersh**
I've noticed that README can be improved in several ways. Let me know what do you think.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,47 +32,81 @@
...
-   * Migration guide from `Prelude`
-   * Haddock with examples for (almost) every function
-     (all examples are tested with [`doctest`](http://hackage.haskell.org/package/doctest))
-   * Documentation regarding [internal module structure]((http://hackage.haskell.org/package/relude/docs/Relude.html))
-   * `relude`-specific [HLint](http://hackage.haskell.org/package/hlint) rules:
-     [`.hlint.yaml`](.hlint.yaml)
+   + Tutorial
+   + Migration guide from `Prelude`
+   + Haddock for every function with examples tested by [`doctest`](http://hackage.haskell.org/package/doctest)
+   + Documentation regarding [internal module structure](http://hackage.haskell.org/package/relude/docs/Relude.html)
```
Comment (vrom911) [inline README.md]: Documentation on?

File: README.md [hunk 2]
```diff
@@ -31,47 +32,81 @@
...
+in `relude`. For detailed documentation of every function together with examples
 and usage, see [_Haddock documentation_](http://hackage.haskell.org/package/relude).
 
+## When to use an alternative prelude? [↑](#structure-of-this-tutorial)
+
+Module with the name `Prelude` is a module imported by default in every Haskell
+source file of your project. If you want to use some data types or functions
+which are not exposed by `Prelude`, you need to import them, adding necessary
+libraries to your project dependencies. Unlike ordinary libraries, alternative
+preludes replace default functions and data types.
```
Comment (vrom911) [inline README.md]: "Replace functions and data types" could sound a bit scary when interpreted incorrectly... Probably somehow rephrase this?

File: README.md [hunk 3]
```diff
@@ -159,47 +194,61 @@ import Relude
...
+* `text`
+* `transformers`
+* `unordered-containers`
 
 If you want to clean up imports after switching to `relude`, you can use
 `relude`-specific [`.hlint.yaml`](.hlint.yaml) configuration for this task.
 
 ### base
 
+Multiple sorting functions are available without imports:
```
Comment (vrom911) [inline README.md]: without imports? probably this is an unnecessary phrase?

Comment (chshersh) [reply README.md]: I just copy-pasted this part. But I agree that part about imports in redundant in the context of preludes. Thanks for noticing!

**Pull request merged**